### PR TITLE
Release/2.0

### PR DESCRIPTION
**Pull request by mts299**
# Scope

pyDARN  2.0 MAJOR release! 
It is  a major release due to the change of interface with reading the files in as it now integrates with pydarnio 
With the addition of 2 new plots! 

## Deadline: Feb 10, 2021

Please indicate if you need more time to test it 

## Approvals

**Number of approvals:** 3

If we get 3 approvals before Feb 10th I will gladly merge the release. 

## Testing

Please focus on the following:
-  [x] io reading 
-  [x] fan plots
-  [x] power plots 
-  [x] range time plots
-  [x] summary plots with slant range
-  [x] documentation: https://pydarn.readthedocs.io/en/release-2.0/  

When testing and approving please indicate the python version and operating system you are using. I would like:
-  [x] Ubuntu
-  [x] OpenSuse
-  [ ] Windows
-  [x] MacOS  

Finally, review the`.zenodo.json` file that your name is correct for the authorlist.

### CODE REVIEW

**Comment by billetd** [comment]
Don't forget returning beam/gate position functionality!

**Comment by carleyjmartin** [comment]
Sorry not a full review yet, I'm just using them now for something else:
I think we're missing a subheading for plot_fan in the readthedocs. I know the .md exists but there's no link to see it.
And I don't know if it's written but we might need a line or two and an example of how to use radar_fov maybe in the radar and hardware subsection?

**Comment by mts299** [comment]
Fan plot is on the main page now, just refresh. 

@billetd can you add a line about radar_fov in the hardware section?

File: docs/about.md [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (carleyjmartin) [inline docs/about.md:1]: Is this file required for read the docs to work, I saw it was removed and them added again? Not an issue though, it's not a blank page on the site anymore.

**Review by carleyjmartin — COMMENTED** [review]
Downloaded and installed using:
`pip3 install git+https://github.com/superdarn/pydarn@release/2.0`

This was all tested on macOS 10.15.5 Catalina.

Methods tested:
- [x] SuperDARNread - no issues with any fitacf 2.5 files that I use
- [x] radar_fov - no issues, noticed that if the `coords` keyword is left blank or spelled wrong, it defaults to 'geo', is this a good feature? Also used this to generate FOV files for an upcoming tool on the website this week and worked brilliantly for that.
- [x] plot_fan - no issue, colourbar overlaps with axis if not resized - I'm aware this is in an issue to figure out if we should allow figure sizes to be read in before or after calling the plotting methods.
- [x] plot_summary slant option -  default is slant range True, both options work but for the following code the axis is too squished, appears to be in 200km ticks but from an arbitrary start number?
```python
import matplotlib.pyplot as plt 
import pydarn
import logging

#Load in file
file = "/Users/carley/Documents/data/20131031.2201.00.sas.fitacf"
SDarn_read = pydarn.SuperDARNRead(file)
fitacf_data = SDarn_read.read_fitacf()
#Plot FOV and show
#a = pydarn.RTP.plot_range_time(fitacf_data, beam_num=5, slant=True)
a = pydarn.RTP.plot_summary(fitacf_data, beam_num=5)
plt.show()
```
Screen Shot 2021-02-02 at 1 20 32 PM

- [x] plot_range_time slant option - no issue, the y-axis isn't squished but same numbers as above (same code with the commented line switched out)
- [x] plot_pwr0_statistics - no issues I can find, but limited knowledge of this method

- Also getting the matplotlib depreciation warnings, but I see that is now an issue.
- Citation warnings all coming up as expected too. 
- No obvious errors or issues in the docs but as I wrote some of that, a fresh pair of eyes on that would be beneficial. 
- My zenodo info is correct
- A .DS_Store has crept in again under pydarn/

In general there are no issues for my use cases or previous testing code that aren't already in issues or being sorted so happy to approve if the squished y-axis is a my code problem and not a pydarn problem

File: docs/about.md (hunk 1 above)
Comment (mts299) [reply docs/about.md:1]: AAAA no it may have got re-added back in with a merge or something

**Comment by mts299** [comment]
@billetd can you look at the `radar_fov` issue @carleyjmartin found?

@carleyjmartin fixed the squashed yaxis issue, let me know if you still get a Deprecation warning

**Comment by carleyjmartin** [comment]
> @carleyjmartin fixed the squashed yaxis issue, let me know if you still get a Deprecation warning

Both good for me now, see what Dan comes up with for the fov thing.

**Comment by billetd** [comment]
Added doc info for `radar_fov`

>radar_fov - no issues, noticed that if the coords keyword is left blank or spelled wrong, it defaults to 'geo', is this a good feature? Also used this to generate FOV files for an upcoming tool on the website this week and worked brilliantly for that.

I think defaulting to geographic is okay, otherwise it would have to assume a date for the AACGM conversion.

**Review by billetd — APPROVED** [review]
I didn't find any issues. Tested on OSX.

**Comment by carleyjmartin** [comment]
> I think defaulting to geographic is okay, otherwise it would have to assume a date for the AACGM conversion.

Is this in the documentation that it will default, I can image someone will misspell "aacgm" and assume they were getting mag coords and not geographic and end up with a problem later down the line?

**Comment by billetd** [comment]
> Is this in the documentation that it will default, I can image someone will misspell "aacgm" and assume they were getting mag coords and not geographic and end up with a problem later down the line?

It is in there, though didn't really think about mispellings. I can add an error message if `coords` keyword is given, but it doesn't match `geo` or `aacgm`?

File: README.md [hunk 2]
```diff
@@ -39,33 +36,27 @@ Please make sure to also read the documentation on [**citing superDARN and pydar
...
 !!! Warning 
-    pyDARN will be removing the IO package in the next release. Please use [pyDARNio](https://github.com/SuperDARN/pyDARNio)
+    pyDARN removed the IO package in the next release. Please use [pyDARNio](https://github.com/SuperDARN/pyDARNio)
 
 ```python
-import pydarn 
+import matplotlib.pyplot as plt
+
+import pydarn
+import pydarnop
```
Comment (ecbland) [inline README.md:45]:
```suggestion
import pydarnio
```

File: README.md [hunk 3]
```diff
@@ -39,33 +36,27 @@ Please make sure to also read the documentation on [**citing superDARN and pydar
 As a quick tutorial on using pydarn to read a non-compressed file: 
 
 !!! Warning 
-    pyDARN will be removing the IO package in the next release. Please use [pyDARNio](https://github.com/SuperDARN/pyDARNio)
+    pyDARN removed the IO package in the next release. Please use [pyDARNio](https://github.com/SuperDARN/pyDARNio)
```
Comment (ecbland) [inline README.md:39]: Just to make this slightly clearer
```suggestion
    pyDARN removed the IO package from this release onwards. Please use [pyDARNio](https://github.com/SuperDARN/pyDARNio)
```

**Review by ecbland — COMMENTED** [review]
Just two small things in the `readme` file

File: README.md (hunk 3 above)
Comment (mts299) [reply README.md:39]: Huh this is outdated

**Review by Devin-Huyghebaert — APPROVED** [review]
OS - Ubuntu 20.04

sites and dates tested:
sas 2020-04-30
mcm 2016-11-24
inv 2015-09-29
hok 2019-04-14
fhw 2017-07-29

- RAWACF & FITACF 3.0 files used

Testing was performed following the documentation provided at https://pydarn.readthedocs.io/en/release-2.0/

- [x] io reading

- tested FITACF read
- tested RAWACF read
- tested compressed file reading

- [ ] fan plots

- tested for FITACF 3.0 files listed above.  Midlatitude radars are cutoff with the default latitude range.  Suggestion: make default lower in latitude

- For southern hemisphere radars setting the "lower" latitude boundary is weird (lowlat option).  Setting -30 sets it at +30, and setting 30 sets it at -30.  Not sure if this is expected behaviour.

- Note:  The location of the range gates were not checked in this review.  Is there a description anywhere for how the range gate locations are determined?

- [ ] power plots

- tested dates and sites listed above.  No issues.

- [ ] range time plots

- tested with dates and sites listed above.  No issues.

- [ ] summary plots with slant range

- tested with dates and sites listed above.  No issues.

- [ ] documentation: https://pydarn.readthedocs.io/en/release-2.0/

- quick read through, looks good.

Overall only a few very minor comments on Fan Plots.

**Comment by mts299** [comment]
@billetd Once finishes the fan suggestions from @Devin-Huyghebaert I will release the code :)

File: pydarn/plotting/fan.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+        # Setup plot
+        # This may screw up references
+        if ax is None:
+            ax = plt.axes(polar=True)
+            if beam_corners_aacgm_lats[0,0] > 0:
+                ax.set_ylim(90, lowlat)
+                ax.set_yticks(np.arange(lowlat, 90, 10))
+            else:
+                ax.set_ylim(-90, -lowlat)
+                ax.set_yticks(np.arange(-lowlat, -90, -10))
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:185]:
```suggestion
                ax.set_ylim(-90, lowlat)
                ax.set_yticks(np.arange(lowlat, -90, -10))
```
I believe this will fix @Devin-Huyghebaert bug found when setting a lowlat value for the southern hemisphere. In this case you now would have to input a -ve value, which I guess makes more sense than it changing the sign for you. I've done it as a suggestion just incase anyone wants to argue for the previous case?

Comment (carleyjmartin) [reply pydarn/plotting/fan.py:185]: By 'I believe' I should say, it does fix it, I have tested it

File: pydarn/plotting/fan.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    def __str__(self):
+        return "This class is static class that provides"\
+                " the following methods: \n"\
+                "   - plot_fan()\n"\
+                "   - return_beam_pos()\n"
+
+    @classmethod
+    def plot_fan(cls, dmap_data: List[dict], ax=None, scan_index: int = 1,
+                       ranges: List = [0,75], boundary: bool = True,
+                       parameter: str = 'v', lowlat: int = 50, cmap: str = None,
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:51]:
```suggestion
                       parameter: str = 'v', lowlat: int = 30, cmap: str = None,
```
Again this is a suggestion because we have the option to change this with lowlat, I don't know if having it as 30 (which I think gets them all in) would be beneficial, would everyone using high lat radars then have to change it so their fov isn't tiny, I guess it's the opposite issue to the low lat radars getting cut off. The suggested change is there anyway!

**Review by carleyjmartin — COMMENTED** [review]
To address the bugs brought up by @Devin-Huyghebaert's review

File: pydarn/plotting/fan.py (hunk 4 above)
Comment (carleyjmartin) [reply pydarn/plotting/fan.py:185]: Actually this has an issue with default now, bear with me.

Comment (carleyjmartin) [reply pydarn/plotting/fan.py:185]:
```suggestion
                ax.set_ylim(-90, -abs(lowlat))
                ax.set_yticks(np.arange(-abs(lowlat), -90, -10))
```
This is the only solution I can think up so that you can have a positive number default relate to the southern hemisphere. This way you can put -30 in and it'll do -30, but if you put 30 in it'll change it to -30. The caveat being you wont be able to plot past 0degrees on a southern hemisphere plot, which I can't see being a problem anyway as it would look ridiculous on a polar plot? As long as we're not going to move to make spherical plots with northern *and* southern fov on I think this will probably suffice?

**Comment by mts299** [comment]
@Devin-Huyghebaert if you can look at the code sometime soon let me know. 
If not I can release the code tomorrow after I quickly test it.
